src/main/java/org/jabref/model/pdf/search/SearchFieldConstants.java

```diff
@@ -3,13 +3,10 @@
 public class SearchFieldConstants {
 
     public static final String PATH = "path";
-    public static final String TITLE = "title";
-    public static final String AUTHOR = "author";
     public static final String CONTENT = "content";
-    public static final String SUBJECT = "subject";
-    public static final String KEYWORDS = "keywords";
-    public static final String KEY = "key";
     public static final String MODIFIED = "modified";
 
-    public static final String[] PDF_FIELDS = new String[]{PATH, TITLE, AUTHOR, CONTENT, SUBJECT, KEYWORDS, KEY, MODIFIED};
+    public static final String[] PDF_FIELDS = new String[]{PATH, CONTENT, MODIFIED};
+
+    public static final String VERSION = "0.2a";
```

Comment:
This shows how changes in indexing should happen. If Fields or Tokenizers change, the index needs to be rebuilt. Changing the version changes the path the index is stored at. If JabRef is opened after such an update, the index is rebuilt at the new location.

What we need is a mechanism to handle out-of-date indices to make sure they don't fill up users file systems.

I would suggest: At startup, silently delete all indices that were built with a different version than what is currently running.
This would probably be the best solution for most users. For people running multiple versions of JabRef at the same time (and if there was an index-update between them) JabRef would rebuild the index each time they switch versions. I think this is acceptable and probably a nieche use-case.

Reply:
Yep, I think this is fine.
What about portable versions?

Reply:
Good point, I need to try what the AppDirs library returns in such a case.

Reply:
But not multiple versions of JabRef but of the Indexer? Since users can turn of the listening to the port and run two instances of JabRef of the same version on the same time...

Reply:
Yes, I introduced a new version number here because I assume the indexing to be updated much less frequently than JabRef. So if the fields stay the same between JabRef 5.4 and 5.5, users can keep the same index and could even run 5.4 and 5.5 at the same time.
Having two instances of JabRef running simultaneously with the same library open could lead to issues where one of them holds the lock on the index. Is this a scenario we have to consider?

Reply:
For the portable versions: I can only check linux and windows. I now log the location of the index when opening a database, @Siedlerchr could you check on mac please?

Reply:
Deletion of the out-of-date indices is in cb1c3935f

Reply:
Works well with the portable versions, index locations are as expected.